### DISCUSSION TITLE
DNS resolution for raspberrypi.local stopped working in 12.0.2

### DISCUSSION THREAD

**Discussion by bongibong** [arendst/Tasmota#16150] [Q&A] [answered]
installed version Tasmota 12.0.2 on several of my sonoff devices.
all devices with that version without any configuration change, reports mqtt connection issue:
MQT: Connect failed to raspberrypi.local:1883, rc -5. Retry in 120 sec
other devices that still on version 11, works fine.
when i set the device mqtt server to the my raspberrypi server IP instead of raspberrypi.local it works fine!
pinging raspberrypi.local from my laptop works.

STATUS 0:
CMD: STATUS 0
18:42:11.889 RSL: STATUS = {"Status":{"Module":29,"DeviceName":"OutLight2","FriendlyName":["OutLight2","OutLight2"],"Topic":"OutLight2","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
18:42:11.896 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"[http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System](http://ota.tasmota.com/tasmota/release/tasmota.bin.gz%22,%22RestartReason%22:%22Software/System) restart","Uptime":"0T00:25:53","StartupUTC":"2022-08-03T17:16:18","Sleep":50,"CfgHolder":4617,"BootCount":45,"BCResetTime":"2021-04-03T14:57:21","SaveCount":1329,"SaveAddress":"FA000"}}
18:42:11.903 RSL: STATUS2 = {"StatusFWR":{"Version":"12.0.2(tasmota)","BuildDateTime":"2022-06-20T12:41:03","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285","CR":"404/699"}}
18:42:11.908 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["bongi",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A000000000000","00000080","00006000","00004000","00000000"]}}
18:42:11.917 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":619,"Free":384,"Heap":26,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"144051","FlashFrequency":40,"FlashMode":3,"Features":["00000809","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","14000020","00000000"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6"}}
18:42:11.924 RSL: STATUS5 = {"StatusNET":{"Hostname":"OutLight2-0287","IPAddress":"10.100.102.3","Gateway":"10.100.102.1","Subnetmask":"255.255.255.0","DNSServer1":"10.100.102.1","DNSServer2":"1.1.1.1","Mac":"24:A1:60:xx:xx:xx","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
18:42:11.927 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"raspberrypi.local","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_10811F","MqttUser":"DVES_USER","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
18:42:11.933 RSL: STATUS7 = {"StatusTIM":{"UTC":"2022-08-03T17:42:11","Local":"2022-08-03T18:42:11","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"05:26","Sunset":"20:25"}}
18:42:11.938 RSL: STATUS10 = {"StatusSNS":{"Time":"2022-08-03T18:42:11"}}
18:42:11.943 RSL: STATUS11 = {"StatusSTS":{"Time":"2022-08-03T18:42:11","Uptime":"0T00:25:53","UptimeSec":1553,"Heap":26,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"bongi","BSSId":"10:13:31:zz:zz:zz","Channel":11,"Mode":"11n","RSSI":66,"Signal":-67,"LinkCount":1,"Downtime":"0T00:00:06"}}}

weblog 4:
00:00:00.001 HDW: ESP8285
00:00:00.050 CFG: Loaded from flash at F9, Count 1330
00:00:00.055 QPC: Count 1
00:00:00.057 CFG: CR 404/699, Busy 0
00:00:00.062 ROT: Mode 1
00:00:00.063 SRC: Restart
00:00:00.065 Project tasmota - OutLight2 Version 12.0.2(tasmota)-2_7_4_9(2022-06-20T12:41:03)
00:00:00.219 WIF: Checking connection...
00:00:00.220 WIF: Attempting connection...
00:00:00.556 WIF: Connecting to AP1 bongi Channel 11 BSSId 10:13:31:23:65:4D in mode 11n as OutLight2-0287...
00:00:01.802 WIF: Checking connection...
00:00:01.803 WIF: Connected
00:00:02.004 HTP: Web server active on OutLight2-0287 with IP address 10.100.102.3
00:00:02.403 WIF: Sending Gratuitous ARP
00:00:02.404 NTP: Sync time...
00:00:02.448 RTC: UTC 2022-08-03T17:44:57, DST 2022-03-27T02:00:00, STD 2022-10-30T03:00:00
18:44:57.000 RTC: Synced by NTP
18:44:58.006 MQT: Attempting connection...
18:44:58.029 DNS: Unable to resolve 'raspberrypi.local'
18:44:58.030 MQT: Connect failed to raspberrypi.local:1883, rc -5. Retry in 10 sec
18:45:01.485 QPC: Reset
18:45:01.946 HTP: Main Menu
18:45:03.223 HTP: Console
18:45:03.384 APP: Boot Count 46
18:45:03.624 CFG: Saved to flash at F8, Count 1331, Bytes 4096
18:45:09.037 MQT: Attempting connection...
18:45:09.065 DNS: Unable to resolve 'raspberrypi.local'
18:45:09.066 MQT: Connect failed to raspberrypi.local:1883, rc -5. Retry in 20 sec
18:45:16.793 WIF: Checking connection...
18:45:30.008 MQT: Attempting connection...
18:45:30.036 DNS: Unable to resolve 'raspberrypi.local'
18:45:30.037 MQT: Connect failed to raspberrypi.local:1883, rc -5. Retry in 30 sec

**Comment by Jason2866** [comment, marked as answer]
Do never ever use `*.local`. It is reserved for mDNS! By doing this you have a name resolution conflict in your network. Tasmota does not use mDNS. 
Solution: change to a other extension (for example `.lan`) and it will work.
In General mDNS is the worst thing ever made...

**Reply by bongibong** [reply]
i didnt add it actually. the raspberry network name is set to 'raspberrypi' only. 
this has been working in all past tasmota versions until now.
from version 12, setting mqtt address to 'raspberrypi.local' or 'raspberrypi' stop working. 
only IP address works now.
